File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Expressions/VarExp.java
package Model.Expressions;
import java.lang.String;

import Exceptions.MyDictionaryException;
import Exceptions.MyStmtException;
import Model.Utils.MyIDictionary;
import Model.Utils.MyIHeap;

public class VarExp extends Exp{
    private String id;
    private int adr;
    public VarExp(String id){this.id=id;}
    public int eval(MyIDictionary<String,Integer> tbl, MyIHeap<Integer> hp) throws MyStmtException {
        try {
            adr=tbl.lookup(id);
            return adr;
        } catch (MyDictionaryException e) {
            throw new MyStmtException(e);
        }
    }
    public String toString(){return id;}
}

File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/C#LabAssignment/sharplab/Model/Expressions/VarExp.cs
using sharplab.Model.Utils;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.Model.Expressions
{
    class VarExp : Exp
    {
        private string id;
        public VarExp(string id) { this.id = id; }
        public override int Eval(IMyDictionary<string,int> tbl)
        { 
            return tbl.LookUp(id);
        }
        public override string ToString() { return id; }
    }
}
